### ISSUE TITLE
Unique inversable flag

### ISSUE THREAD

**Issue by PowerKiKi** [Ecodev/natural#8]
A flag value must always be unique, meaning it can only be selected exactly 1 time per group. Because having a flag multiple times make no sense, because the value is always the same.

Also a flag must be configurable as "inversed", meaning that the flag value is set when the flag is **not** selected, and the flag value is removed when the flag is selected. This is to express "include archive" flag, which by default filter out archived items, without having to be visible in the search bar. Once the "include archive" is selected, it appears in the search bar and it stops filtering out archived items. So the value is not present anymore.

@sambaptista do you agree with the inversable thing ?

**Comment by PowerKiKi** [comment]
Re-reading everything, I am not so sure about the overriding thing. The usage seems way too brittle, because you must configure two things in two (very) different places in your project and you must have a matching name, otherwise nothing work. I can see easily how we will forget about that in future maintenance and breaks the app in a non-obvious way.

 Also this would make it blocked by #2.

Having a single boolean on `FlagFacet` like so:

```ts
/**
 * Facet that is only a flag (set or unset)
 */
export interface FlagFacet extends BasicFacet {

    /**
     * The value to be returned when the flag is set
     */
    condition: any;

    /**
     * Whether the value is set when the flag exist or does not exist
     */
    inversed: boolean;
}
```

makes it not only much easier to implement in this lib, but also much easier to use and maintain in projects.

**Comment by sambaptista** [comment]
The reason I've argued about overriding, is that selection.visible and facet.visible are nice to have features and they could be done out of the usecase of this issue. 

The overriding system, combined with visible attributes would be generic and powerfull enough to satisfy the usecase of this issue. 

Adding facet.inversed attribute is an additional and specific feature for this usecase that works more like a shortcut. For sure it should remove developper errors... so, let's say I agree.

With overriding system the preselections #2 block this ticket. Without preselections #2 it's not blocked.

**Comment by PowerKiKi** [comment]
Unique constraint was not implemented because of lack of time, but it could/should be added later.
